Receive a block to ActionController::Parameters.to_h

bobf opened the pull request:
### Summary

Allow `ActionController::Parameters.to_h` to receive a block to provide parity with `Hash#to_h`. The provided block receives `key, value` and yields a two-element array/keypair which can be transformed in the resulting Hash.

### Other Information

https://ruby-doc.org/core-2.7.5/Hash.html#method-i-to_h

```
to_h → hsh or new_hashc
to_h {|key, value| block } → new_hash
Returns self. If called on a subclass of Hash, converts the receiver to a Hash object.

If a block is given, the results of the block on each pair of the receiver will be used as pairs.
```

bobf committed:
Receive a block to ActionController::Parameters.to_h

Allow `ActionController::Parameters.to_h` to receive a block to provide
parity with `Hash#to_h`. The provided block recieves `key, value` and
yields a two-element array/keypair which can be transformed in the
resulting Hash.

https://ruby-doc.org/core-2.7.5/Hash.html#method-i-to_h

rafaelfranca commented on actionpack/test/controller/parameters/to_h_test.rb:
```diff
@@ -0,0 +1,34 @@
+require "abstract_unit"
+require "action_controller/metal/strong_parameters"
+
+class ParametersToHTest < ActiveSupport::TestCase
```
let's move those tests to the same file as all the other parameters methods.

rafaelfranca commented on actionpack/test/controller/parameters/to_h_test.rb:
```diff
@@ -0,0 +1,34 @@
+  # rubocop:disable HashTransformKeys
+  test "receives a block and transforms keys" do
+    params_hash = @params.to_h { |key, value| [:"#{key}_modified", value] }
+    assert_equal params_hash.keys, %w(name_modified age_modified location_modified)
```
Let's add assertions to make sure those parameters are not permitted.

bobf commented:
@rafaelfranca Thanks for the feedback - I have moved the tests into `actionpack/test/controller/parameters/mutators_test.rb` and added a test to ensure that `ActionController::UnfilteredParameters` is raised.

I wanted to re-use `@params` but they were not very suitable for testing (only one key in the root hash, values were not too easy to apply a simple transformation for testing).

Let me know if you need any further adjustments.
